lib/veri_helpers.ml

```diff
@@ -0,0 +1,47 @@
+open Core_kernel.Std
+open Bap.Std
+open Bap_traces.Std
+
+module Dis = Disasm_expert.Basic
+
+(** [name_of_insns insns] - returns a name of last instruction in [insns]  *)
+let name_of_insns insns = 
+  match List.hd (List.rev insns) with
+  | None -> Or_error.error_string "instruction hasn't name"
+  | Some (_, insn) -> Ok (Insn.(name (of_basic insn)))
```

Comment:
gosh, basic instruction also has a name, there is no need to transform it to regular instructions, just to get a name. First of all, `of_basic` is not a free operation.